### PR TITLE
Feature/6

### PR DESCRIPTION
**Pull request by theAmbientOrange**
Sign-In Functionality

- Uses basic Firebase Authentication -> Email + Password combination
- Adds views and corresponding controllers for existing/new users 😄

### CODE REVIEW

File: BigTuna/Base.lproj/Main.storyboard [hunk 1]
```diff
@@ -89,10 +89,208 @@
...
+                                                        <rect key="frame" x="0.0" y="139" width="254" height="42.666666666666657"/>
+                                                        <nil key="textColor"/>
+                                                        <fontDescription key="fontDescription" type="system" pointSize="14"/>
+                                                        <textInputTraits key="textInputTraits" secureTextEntry="YES"/>
+                                                    </textField>
+                                                    <button opaque="NO" contentMode="scaleToFill" contentHorizontalAlignment="center" contentVerticalAlignment="center" buttonType="roundedRect" lineBreakMode="middleTruncation" translatesAutoresizingMaskIntoConstraints="NO" id="VO1-19-plc">
+                                                        <rect key="frame" x="0.0" y="185.66666666666669" width="254" height="42.333333333333343"/>
+                                                        <state key="normal" title="sign upp"/>
```
Comment (wgwong) [inline BigTuna/Base.lproj/Main.storyboard]: @theAmbientOrange  nit: typo

File: BigTuna/View Controllers/LoginViewController.swift [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        default:
+            break;
+        }
+    }
+    
+    
+    @IBAction func signInButtonPressed(_ sender: UIButton) {
+        guard let email = signInEmailTextField.text,
+            let password = signInPasswordTextField.text else {
+                self.displayErrorWithMessage(error: "Please enter all fields.")
```
Comment (wgwong) [inline BigTuna/View Controllers/LoginViewController.swift]: nit: move string to constants

File: BigTuna/View Controllers/LoginViewController.swift [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+            }
+        }
+    }
+    
+    @IBAction func signUpButtonPressed(_ sender: UIButton) {
+        guard let _ = signUpUsernameTextField.text,
+            let email = signUpEmailTextField.text,
+            let password = signUpPasswordTextField.text,
+            let _ = signUpConfirmPasswordTextField.text else {
+                self.displayErrorWithMessage(error: "Please enter all fields.")
```
Comment (wgwong) [inline BigTuna/View Controllers/LoginViewController.swift]: move string to constants

File: BigTuna/Base.lproj/Main.storyboard (hunk 1 above)
Comment (theAmbientOrange) [reply BigTuna/Base.lproj/Main.storyboard]: Fixed 😄

**Comment by theAmbientOrange** [comment]
@cyt3ea 
We do not currently have password validation or a forgot password page. We could add that in the future but I would de-prioritize those tasks for now for MVP 👍👍

**Comment by theAmbientOrange** [comment]
@cyt3ea 
Also I have fixed the textfield/keyboard UX, please review again 😄

**Pull request merged**